plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -50,6 +171,155 @@ public void testVarbinary()
+                .addRoundTrip("varchar", "CAST('攻殻機動隊' AS varchar)", createUnboundedVarcharType(), "CAST('攻殻機動隊' AS varchar)")
+                .addRoundTrip("varchar", "CAST('攻殻機動隊' AS varchar)", createUnboundedVarcharType(), "CAST('攻殻機動隊' AS varchar)")
+                .addRoundTrip("varchar", "CAST('😂' AS varchar)", createUnboundedVarcharType(), "CAST('😂' AS varchar)")
+                .addRoundTrip("varchar", "CAST('Ну, погоди!' AS varchar)", createUnboundedVarcharType(), "CAST('Ну, погоди!' AS varchar)")
+                .addRoundTrip("varchar", "'text_f'", createUnboundedVarcharType(), "CAST('text_f' AS varchar)")
+                .execute(getQueryRunner(), trinoCreateAndInsert(getSession(), "test_varchar_unbounded"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_varchar_unbounded"));
+    }
+
+    @Test
+    public void testVarcharMapping()
+    {
+        // testing mapping varchar > 4000 -> varchar(max)
+        SqlDataTypeTest.create()
+                .addRoundTrip("varchar(4001)", "'text_c'", createUnboundedVarcharType(), "CAST('text_c' AS varchar)")
+                .execute(getQueryRunner(), trinoCreateAndInsert(getSession(), "test_varchar_mapping"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_varchar_mapping"));
+    }
+
+    @Test
+    public void testChar()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("char(1)", "NULL", createCharType(1), "CAST(NULL AS char(1))")
+                .addRoundTrip("char(10)", "'text_a'", createCharType(10), "CAST('text_a' AS char(10))")
+                .addRoundTrip("char(255)", "'text_b'", createCharType(255), "CAST('text_b' AS char(255))")
+                .addRoundTrip("char(4001)", "'text_c'", createCharType(4001), "CAST('text_c' AS char(4001))")
+                .execute(getQueryRunner(), onRemoteCreateAndInsert("test_char"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("char(1)", "NULL", createCharType(1), "CAST(NULL AS char(1))")
+                .addRoundTrip("char(10)", "'text_a'", createCharType(10), "CAST('text_a' AS char(10))")
+                .addRoundTrip("char(255)", "'text_b'", createCharType(255), "CAST('text_b' AS char(255))")
+                .addRoundTrip("char(5)", "CAST('攻殻機動隊' AS char(5))", createCharType(5), "CAST('攻殻機動隊' AS char(5))")
+                .addRoundTrip("char(32)", "CAST('攻殻機動隊' AS char(32))", createCharType(32), "CAST('攻殻機動隊' AS char(32))")
+                .addRoundTrip("char(20)", "CAST('😂' AS char(20))", createCharType(20), "CAST('😂' AS char(20))")
+                .addRoundTrip("char(77)", "CAST('Ну, погоди!' AS char(77))", createCharType(77), "CAST('Ну, погоди!' AS char(77))")
+                .execute(getQueryRunner(), trinoCreateAndInsert(getSession(), "test_char"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_char"));
+    }
+
+    @Test
+    public void testCharMapping()
```

Comment:
as above -- not a meaningful test name, and this can be avoided by merging this into test method above, and then perhaps slicing into testSqlServerChar, testSqlServerNchar, testTrinoChar